rust/lance-encoding/src/encoder.rs

```diff
@@ -252,9 +270,45 @@ impl CoreArrayEncodingStrategy {
     }
 }
 
+// check whether we want to use dictionary encoding or not
+// by applying a threshold on cardinality
+// returns true if cardinality < 100
+// The choice to use 100 is just a heuristic for now
+fn check_dict_encoding(arrays: &[ArrayRef]) -> bool {
```

Comment:
Hmm...in retrospect this also means we will encode all string arrays that have length less than 100 as dictionary too.  I guess this is probably harmless but maybe we should only use dictionary encoding if there are less than `min(100, num_rows)` values?

Reply:
Won't this condition be satisfied regardless? Since the number of unique values must be <= `num_rows` in any case

Reply:
Right, I guess I am saying it should be strictly `<` and not `<=`.